sdk/containerregistry/container-registry/samples/v1-beta/typescript/src/uploadCustomManifest.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Uploads a manifest with custom manifest type, in this case a manifest list.
+ */
+
+import { ContainerRegistryBlobClient } from "@azure/container-registry";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+import { Readable } from "stream";
+dotenv.config();
+
+async function main() {
+  // endpoint should be in the form of "https://myregistryname.azurecr.io"
+  // where "myregistryname" is the actual name of your registry
+  const endpoint = process.env.CONTAINER_REGISTRY_ENDPOINT || "<endpoint>";
+  const repository = process.env.CONTAINER_REGISTRY_REPOSITORY || "library/hello-world";
+  const client = new ContainerRegistryBlobClient(
+    endpoint,
+    repository,
+    new DefaultAzureCredential()
+  );
+
+  const mediaType = "application/vnd.docker.distribution.manifest.list.v2+json";
+
+  const manifest = Readable.from(
+    JSON.stringify({
+      schemaVersion: 2,
+      mediaType,
+      manifests: [
+        {
+          mediaType: "application/vnd.docker.distribution.manifest.v2+json",
+          digest: "sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f",
+          size: 7143,
+          platform: {
+            architecture: "ppc64le",
+            os: "linux",
+          },
+        },
+        {
+          mediaType: "application/vnd.docker.distribution.manifest.v2+json",
+          digest: "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
+          size: 7682,
+          platform: {
+            architecture: "amd64",
+            os: "linux",
+            features: ["sse4"],
+          },
+        },
+      ],
+    })
+  );
+
+  await client.uploadManifest(manifest, { mediaType });
```

Comment:
curious how would both mediaType and tag look like?

Reply:
The options bag is a JavaScript object (kind of like a map or dictionary), it would look like

```ts
await client.uploadManifest(manifest, { mediaType, tag });
```

This syntax is a shorthand for when the key of the object is the same as the variable name. You could also write:

```js
await client.uploadManifest(manifest, {
  mediaType: mediaType,
  tag: tag
});
```

This is a common pattern in all our SDKs and in JavaScript in general, so I don't think we necessarily need to explicitly set the tag in this example (although it's probably the more common case?).